src/main/java/org/stratumproject/fabric/tna/behaviour/upf/FabricUpfTranslator.java

```diff
@@ -672,6 +748,48 @@ public FlowRule interfaceToFabricEntry(UpfInterface upfInterface, DeviceId devic
                 .build();
     }
 
+    public MeterRequest upfMeterToFabricMeter(UpfMeter upfMeter, DeviceId deviceId, ApplicationId appId)
+            throws UpfProgrammableException {
+        final PiMeterId meterId;
+        if (upfMeter.type().equals(UpfEntityType.SESSION_METER)) {
+            meterId = FABRIC_INGRESS_UPF_SESSION_METER;
+        } else if (upfMeter.type().equals((UpfEntityType.APPLICATION_METER))) {
+            meterId = FABRIC_INGRESS_UPF_APP_METER;
+        } else {
+            // I should never reach this point.
+            throw new UpfProgrammableException("Unknown UPF meter type. I should never reach this point! " + upfMeter);
+        }
+        MeterRequest.Builder meterRequest = DefaultMeterRequest.builder()
+                .forDevice(deviceId)
+                .fromApp(appId)
+                .withScope(MeterScope.of(meterId.id()))
+                .withUnit(Meter.Unit.BYTES_PER_SEC)
+                .withIndex((long) upfMeter.cellId());
+        if (upfMeter.isReset()) {
+            return meterRequest.remove();
+        } else {
+            Collection<Band> bands = Lists.newArrayList();
+            if (upfMeter.committedBand().isPresent()) {
+                bands.add(upfMeter.committedBand().get());
+            } else {
+                bands.add(DefaultBand.builder()
+                                  .ofType(Band.Type.MARK_YELLOW)
+                                  .withRate(0L).burstSize(0L)
```

Comment:
I'm almost sure this will be rejected, stratum will reject it for sure. Need a little bit of investigation.